fsm.go

```diff
@@ -249,6 +263,32 @@ func (f *FSM) Cannot(event string) bool {
 	return !f.Can(event)
 }
 
+// ReadData returns the value stored in data
+func (f *FSM) ReadData(key string) interface{} {
```

Comment:
Could these be renamed to `Metadata()` and `SetMetadata()`. Any other name suggestions?

Reply:
yeah , we can rename them to `GetMetaData()` and `SetMetadata()`. you agree?

Reply:
I specifically did not want to use "get" its not idiomatic in Golang.